mmv1/products/integrationconnectors/Connection.yaml

```diff
@@ -0,0 +1,1132 @@
+            properties:
+              - !ruby/object:Api::Type::String
+                name: "secretVersion"
+                required: true
+                description: |
+                  The resource name of the secret version in the format,
+                  format as: projects/*/secrets/*/versions/*.
+          - !ruby/object:Api::Type::Array
+            name: "scopes"
+            description: |
+              Scopes the connection will request when the user performs the auth code flow.
+            item_type: Api::Type::String
+          - !ruby/object:Api::Type::Boolean
+            name: "enablePkce"
+            description: |
+              Whether to enable PKCE when the user performs the auth code flow.
+          - !ruby/object:Api::Type::String
+            name: "authUri"
+            description: |
+              Auth URL for Authorization Code Flow.
+        exactly_one_of:
+          - auth_config.0.user_password
+          - auth_config.0.oauth2_jwt_bearer
+          - auth_config.0.oauth2_client_credentials
+          - auth_config.0.ssh_public_key
+          - auth_config.0.oauth2_auth_code_flow
+
+  - !ruby/object:Api::Type::NestedObject
+    name: "lockConfig"
+    description: |
+      Determines whether or no a connection is locked. If locked, a reason must be specified.
+    properties:
+      - !ruby/object:Api::Type::Boolean
+        name: "locked"
+        description: |
+          Indicates whether or not the connection is locked.
+        required: true
+      - !ruby/object:Api::Type::String
+        name: "reason"
+        description: |
+          Describes why a connection is locked.
+  - !ruby/object:Api::Type::Array
+    name: "destinationConfig"
+    api_name: "destinationConfigs"
+    description: |
+      Define the Connectors target endpoint.
+    item_type: !ruby/object:Api::Type::NestedObject
+      properties:
+        - !ruby/object:Api::Type::String
+          name: "key"
+          description: |
+            The key is the destination identifier that is supported by the Connector.
+          required: true
+        - !ruby/object:Api::Type::Array
+          name: "destination"
+          api_name: "destinations"
+          description: |
+            The destinations for the key.
+          item_type: !ruby/object:Api::Type::NestedObject
+            properties:
+              - !ruby/object:Api::Type::Integer
+                name: "port"
+                description: |
+                  The port is the target port number that is accepted by the destination.
+              - !ruby/object:Api::Type::String
+                name: "serviceAttachment"
+                description: |
+                  PSC service attachments. Format: projects/*/regions/*/serviceAttachments/*
+              - !ruby/object:Api::Type::String
+                name: "host"
+                description: |
+                  For publicly routable host.
+  - !ruby/object:Api::Type::String
+    name: "serviceAccount"
```

Comment:
looks like we need `default_from_api` for `service_account` as the API still return the default service account even if we don't specify it

Reply:
That's correct.